tests/js/client/aql/aql-mdi.js

```diff
@@ -536,6 +544,78 @@ function optimizerRuleMdi2dIndexTestSuite() {
         assertEqual([0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1], res);
       }
     },
+
+    testInRangeMixed: function () {
+      const queries = [
+        aql`FOR d IN ${col} FILTER IN_RANGE(d.x, 0, 1, false, true) AND 0 <= d.y AND d.y <= 5 RETURN d.x`,
+        aql`FOR d IN ${col} FILTER 0 <= d.y AND d.y <= 5 AND IN_RANGE(d.x, 0, 1, false, true) RETURN d.x`,
+      ];
```

Comment:
```suggestion
      const queries = [
        aql`FOR d IN ${col} FILTER IN_RANGE(d.x, 0, 1, false, true) AND 0 <= d.y AND d.y <= 5 RETURN d.x`,
        aql`FOR d IN ${col} FILTER 0 <= d.y AND d.y <= 5 AND IN_RANGE(d.x, 0, 1, false, true) RETURN d.x`,
        aql`FOR d IN ${col} FILTER IN_RANGE(d.y, 0, 5, true, true) AND 0 < d.x AND d.x <= 1 RETURN d.x`,
        aql`FOR d IN ${col} FILTER 0 < d.x AND d.x <= 1 AND IN_RANGE(d.y, 0, 5, true, true) RETURN d.x`,
      ];
```
lets add more test cases here!